src/SAWScript/Prover/MRSolver/Evidence.hs

```diff
@@ -0,0 +1,189 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{- |
+Module      : SAWScript.Prover.MRSolver.Evidence
+Copyright   : Galois, Inc. 2023
+License     : BSD3
+Maintainer  : westbrook@galois.com
+Stability   : experimental
+Portability : non-portable (language extensions)
+
+This module defines multiple outward facing components of MRSolver, most
```

Comment:
I feel like the evidence is the least important part of what is currently in this file. Maybe it should be called something else? Also, why is it now separate from Term.hs?

Reply:
I think my motivation to make this separate from `Term.hs` is generally because I prefer smaller, easier to digest Haskell files over larger files containing many separate linked ideas. This set of definitions seem like a linked idea, so I decide to break them out – though I realized I never actually explained how they were linked. I added a comment at the top now but here it is reproduced here:
```
Note: In order to avoid circular dependencies, the 'FunAssump' type and its
dependents in this file ('Refnset' and 'MREvidence') are given a type
parameter `t` which in practice always be 'TheoremNonce' from `Value.hs`.
The reason we cannot just import `Value.hs` here directly is because the
'Refnset' type is used in `Value.hs` - specifically, in the 'VRefnset'
constructor of the 'Value' datatype.
```

I am not sold on the name `Evidence.hs`, though. My thoughts were: (1) at some point that `Evidence` datatype *will* needed to be fleshed out, at which point that section of the file will get much bigger, and (2) the `FunAssump` and `Refnset` datatypes both share the type parameter described above and are both used as key components in `Evidence`, so `Evidence` might be a good way to center the file. But, (1) is not the case now and for (2), you could make the argument that `FunAssump` or `Refnset` are just as central.

So as for the name, up to you. If you think strongly it should be merged back into `Term.hs` that's fine too, I'm just worried about these files having too many parts which can make them harder to read.